### ISSUE TITLE
Process output should be convertable to string

### ISSUE THREAD

**Issue by ForNeVeR** [Pash-Project/Pash#400]
In PowerShell I can do the following:

```
PS> cmd /c where cmd
C:\Windows\System32\cmd.exe
PS> $(cmd /c where cmd).Split('.')[0]
C:\Windows\System32\cmd
```

In Pash I cannot:

```
PASH> whereis ls
ls: /nix/store/5vrf2s2qj8q20vv6wskqkgzdz8h8g4qf-system-path/bin/ls
PASH> $(whereis ls).Split(':')
ls: /nix/store/5vrf2s2qj8q20vv6wskqkgzdz8h8g4qf-system-path/bin/ls
Cannot invoke a method of a NULL expression
  +CategoryInfo: InvalidOperation, Reason: PSInvalidOperationException
  +FullyQualifiedErrorId: InvalidOperation
```

Note that Pash echoes the command output but does not calls methods on it.

**Comment by sburnicki** [comment]
I think the problem is that `whereis` is a local command which by default outputs to the terminal directly and doesn't write the result to pipeline. It does that only in some cases and then there is a `WriteSideEffectsToPipeline` flag set. This basically happens when the `ExecutionVisitor`'s `EvaluateAst` method is called.

For simple local program execution this behavior is wanted, as always writing the program's result to a pipeline would result in loss of output colors and formatting, and sometimes even in functionality.

We'd therefore need to identify the cases when and where the flag is set to true or false and make sure that the flag is set if there is a member invocation expression following.

I'm afraid this topic is currently fairly intransparent and related to the discussion in issue #268.

**Comment by ForNeVeR** [comment]
Yes, I think you're right, that definitely involves this horrible `WriteSideEffectsToPipeline` flag.
